hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/expression/json/JsonValueFunction.java

```diff
@@ -226,6 +240,7 @@ public void readData(final ObjectDataInput in) throws IOException {
         super.readData(in);
         this.onEmpty = SqlJsonValueEmptyOrErrorBehavior.valueOf(in.readString());
         this.onError = SqlJsonValueEmptyOrErrorBehavior.valueOf(in.readString());
+        prepareCache();
```

Comment:
Can't we serialize the `constantPathCache` here? Isn't it the reason why we made the `JsonPath` serializable in json-surfer?

Reply:
I thought it will be serialized in ```com.hazelcast.jet.impl.AbstractJobProxy#doSubmitJob``` with ```serializationService().toData(jobDefinition)```. What am I missing?

Reply:
I believe I understood my mistake, added serialization.

Reply:
I will benchmark those cases. We can:
- preserialize constantPathCache
- preserialize pathCache

but those will contain Java Serialization, I want to make sure that ser/deser is faster.

Reply:
Without Java Serialization the latency is better:
![image](https://user-images.githubusercontent.com/57556371/160615914-2fa0ade9-8037-4c98-a501-acd55331e499.png)
![image](https://user-images.githubusercontent.com/57556371/160615936-1da351ee-c3bf-49a8-acfb-9320ab94681c.png)
If you approve this I will revert Serialization in json surfer project, it is not needed.